src/Compiler/Service/service.fs

```diff
@@ -1450,7 +1461,20 @@ type FSharpChecker
         let canInvalidateProject = defaultArg canInvalidateProject true
         let userOpName = defaultArg userOpName "Unknown"
 
-        backgroundCompiler.FindReferencesInFile(fileName, options, symbol, canInvalidateProject, userOpName)
+        node {
+            if fastFindReferences <> Some true then
+                return! backgroundCompiler.FindReferencesInFile(fileName, options, symbol, canInvalidateProject, userOpName)
+            else
+                let! parseResults = backgroundCompiler.GetBackgroundParseResultsForFileInProject(fileName, options, userOpName)
+
+                if
+                    parseResults.ParseTree.Identifiers |> Set.contains symbol.DisplayNameCore
```

Comment:
Actually it does. It seems in some cases it works differently when starting at the attribute usage (without suffix). I'll have a look at that.

Reply:
So it doesn't matter whether or not you click on a symbol with Attribute omitted or not, but it wasn't working when you'd click on the attribute usage - because it ended up being a `FSharpMemberOrFunctionOrValue` (the attribute's constructor) instead of a an `FSharpEntity`. Fixed that.